src/db.js

```diff
@@ -0,0 +1,50 @@
+import Dexie from 'dexie'
+import _ from 'lodash'
+
+const db = new Dexie('EM')
+
+const initHelpers = async () => {
+  const staticHelpersExist = await db.helpers.get({ id: 'EM' })
+  if (!staticHelpersExist) {
+    await db.helpers.add({ id: 'EM' })
+  }
+}
+
+const initDB = async () => {
+  await db.version(1).stores({
+    thoughtIndex: 'id, value, *contexts, created, lastUpdated',
+    contextIndex: 'id, *context, lastUpdated',
+    helpers: 'id, cursor, lastUpdated, recentlyEdited, schemaVersion'
+  })
+  await initHelpers()
+}
+
+export const updateThought = async (id, thought) => db.thoughtIndex.put({ id, ...thought })
+export const updateThoughtIndex = async thoughtIndexMap => {
+  const thoughtsArray = Object.keys(thoughtIndexMap).map(key => ({ ...thoughtIndexMap[key], id: key }))
+  return db.thoughtIndex.bulkPut(thoughtsArray)
+}
+export const deleteThought = async id => db.thoughtIndex.delete(id)
+export const getThought = async id => db.thoughtIndex.get(id)
+export const getThoughtIndex = async () => {
+  const contextIndexMap = await db.thoughtIndex.toArray()
```

Comment:
This should be called `thoughtIndexMap`.

Reply:
Sorry. That was my bad. Fixing it now